### PR TITLE
[bitnami/thanos] Automatically apply query-frontend's ingress hostname to alert.queryURL

### LINKED ISSUE #16401: [thanos] Handle alert.query-url parameter automatically
### Name and Version

bitnamicharts/thanos 12.5.1

### What is the problem this feature will solve?

Add `--alert.query-url` automatically to `query` and `ruler` when `queryFrontend` have `.Values.ingress.enabled` and configured properly

### What is the feature you are proposing to solve the problem?

It will allow provide links in Thanos UI properly and also will generate proper alerts source label which will be send to alertmanager by `ruler`

### What alternatives have you considered?

_No response_

### PR DESCRIPTION
**Pull request by fabriciolos**
### Description of the change

Added `.queryFrontend.ingress.overrideAlertQueryURL` as an option to automatically override `alert.queryURL` to Query-Frontend's ingress hostname on both Ruler and Query.

### Benefits

When using query-frontend through an ingress, there is the need to include the flag  `--alert.queryURL` on both ruler and query pods, in order for the correct link for rules/expressions to be checked on Rules/Alerts tabs, and also on Alertmanager's alerts.

### Possible drawbacks

Only works with the value of `queryFrontend.ingress.hostname`.
Fixed "http://" url without port.

### Applicable issues

- fixes #16401

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/thanos/templates/_helpers.tpl [hunk 1]
```diff
@@ -158,13 +158,17 @@ Return the Thanos Ruler configuration configmap.
 Return the queryURL used by Thanos Ruler.
 */}}
 {{- define "thanos.ruler.queryURL" -}}
+{{- if and .Values.queryFrontend.enabled .Values.queryFrontend.ingress.enabled .Values.queryFrontend.ingress.hostname  .Values.queryFrontend.ingress.overrideAlertQueryURL -}}
```
Comment (andresbono) [inline bitnami/thanos/templates/_helpers.tpl:161]: Small suggestion here:

```suggestion
{{- if and .Values.queryFrontend.enabled .Values.queryFrontend.ingress.enabled .Values.queryFrontend.ingress.hostname .Values.queryFrontend.ingress.overrideAlertQueryURL -}}
```

**Review by andresbono — COMMENTED** [review]
A couple of really minor suggestions, the rest LGTM. Thank you for your contribution with this new feature!!

**Comment by fabriciolos** [comment]
@andresbono all done! Thanks for the feedback.

**Review by andresbono — APPROVED** [review]

**Comment by andresbono** [comment]
Merged! Thank you again for your contribution!

**Pull request merged**